src/sql/src/plan/lowering.rs

```diff
@@ -722,7 +722,12 @@ impl HirScalarExpr {
                                 // Record input arity here so that any group_keys that need to mutate get_inner
                                 // don't add those columns to the aggregate input.
                                 let input_arity = get_inner.typ().arity();
-                                let mut group_key = Vec::new();
+                                let mut group_key = col_map
```

Comment:
> The reduction that computes the window function must be keyed on the columns from the outer context.
Can you add this as a comment here to explain what is happening? 

Why is it necessary to group by the columns from the outer context?

Reply:
Good point. I took it for granted but it's not obvious at all. Just added a comment explaining why.